packages/neo-one-client/src/__tests__/Client.test.js

```diff
@@ -0,0 +1,235 @@
+  });
+
+  test('Client constructor throws error on null client', () => {
+    function testError() {
+      client = new Client({});
+    }
+
+    expect(testError).toThrow(new Error('At least one provider is required'));
+  });
+
+  test('Client constructor throws error on mismatched provider type', () => {
+    function testError() {
+      client = new Client({ test: { type: 'invalid' } });
+    }
+
+    expect(testError).toThrow(
+      new Error('Provider keys must be named the same as their type'),
+    );
+  });
+
+  test('get providers', () => {
+    const result = client.providers;
+    // $FlowFixMe
+    expect(result).toEqual(client._providers$.value);
+  });
+
+  test('selectAccount throws error on unknown account', async () => {
+    const accountID = { network: 'fakeNet', address: 'fakeAddr' };
+
+    // $FlowFixMe
+    providers[0].getAccounts = jest.fn(() => providers[0].accounts$);
+    // $FlowFixMe
+    providers[1].getAccounts = jest.fn(() => providers[1].accounts$);
+
+    const result = client.selectAccount(accountID);
+
+    await expect(result).rejects.toEqual(
+      new UnknownAccountError(accountID.address),
+    );
+  });
+
+  test('selectAccount', async () => {
+    const accountID = id2;
+    const expectedProvider = provider2;
+    // $FlowFixMe
+    providers[0].getAccounts = jest.fn(() => providers[0].accounts$);
+    // $FlowFixMe
+    providers[1].getAccounts = jest.fn(() => providers[1].accounts$);
+    // $FlowFixMe
+    providers[1].selectAccount = jest.fn(async () => Promise.resolve({}));
+    // $FlowFixMe
+    expect(client._selectedProvider$.value).toEqual(provider1);
+    await client.selectAccount(accountID);
+    // $FlowFixMe
+    expect(client._selectedProvider$.value).toEqual(expectedProvider);
+  });
+
+  test('getAccounts', () => {
+    const expected = [account1, account2];
+    // $FlowFixMe
+    providers[0].getAccounts = jest.fn(() => providers[0].accounts$);
+    // $FlowFixMe
+    providers[1].getAccounts = jest.fn(() => providers[1].accounts$);
+
+    const result = client.getAccounts();
+    expect(result).toEqual(expected);
+  });
+
+  test('getNetworks', () => {
+    const expected = [network1, network2];
+    // $FlowFixMe
+    providers[0].getNetworks = jest.fn(() => providers[0].networks$);
+    // $FlowFixMe
+    providers[1].getNetworks = jest.fn(() => providers[1].networks$);
+
+    const result = client.getNetworks();
+    expect(result).toEqual(expected);
+  });
+
+  test('inject', () => {
+    const provider3 = { type: 'inject' };
+    // $FlowFixMe
+    expect(client._selectedProvider$.value).toEqual(provider1);
+    // $FlowFixMe
+    client.inject(provider3);
+    // $FlowFixMe
+    expect(client._selectedProvider$.value).toEqual(provider3);
+    expect(client.providers).toEqual({
+      test1: provider1,
+      test2: provider2,
+      inject: provider3,
+    });
+  });
+
+  test('static inject', () => {
+    const provider3 = { type: 'inject' };
```

Comment:
`const provider3 = ({ type: 'inject' }: $FlowFixMe);` and remove the ones below.